### PR TITLE
react-transition-group: pass nodeRef so findDOMNode isn't used

### PR DESCRIPTION
**Pull request by snowystinger**
https://github.com/adobe/react-spectrum/pull/1108/ is failing due to Adobe CLA, opening a new PR to fix that
@mischnic :
Part of #779

There are more problems with strict mode (which is why I didn't enable strict mode in tests or Storybook yet), this is just fixing the react-transition-group and findDOMNode problem.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx [hunk 1]
```diff
@@ -30,8 +31,19 @@ function TooltipTrigger(props: SpectrumTooltipTriggerProps) {
     offset = DEFAULT_OFFSET
   } = props;
 
+  let tooltipRef = useRef();
   let [trigger, tooltip] = React.Children.toArray(children);
 
+  // @ts-ignore
```
Comment (dannify) [inline packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx:37]: there is a lot of ignoring here. What are we ignoring?

File: packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx [hunk 2]
```diff
@@ -30,7 +31,22 @@ function TooltipTrigger(props: SpectrumTooltipTriggerProps) {
...
   } = props;
 
-  let [trigger, tooltip] = React.Children.toArray(children);
+  let tooltipRef = useRef();
+
+  let [trigger, tooltip] = React.Children.toArray(children) as [ReactElement, ReactElement];
+
+  // this does mean that we can't have some element wrapping the Tooltip
+  // we'd need to do it through context if we wanted to allow for that
+  let tooltipWithRef = React.cloneElement(tooltip, {
```
Comment (devongovett) [inline packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx:40]: Yeah we should avoid `cloneElement`. Someone might do something like `<MyFancyTooltip>` that wraps `<Tooltip>` and not pass through the ref...

File: packages/@react-spectrum/overlays/src/Modal.tsx [hunk 3]
```diff
@@ -35,9 +35,11 @@ function Modal(props: ModalProps, ref: DOMRef<HTMLDivElement>) {
   let domRef = useDOMRef(ref);
   let {styleProps} = useStyleProps(props);
 
+  let underlayRef = useRef(null);
+
   return (
-    <Overlay {...otherProps}>
-      <Underlay />
+    <Overlay {...otherProps} nodeRef={underlayRef}>
+      <Underlay ref={underlayRef} />
```
Comment (ktabors) [inline packages/@react-spectrum/overlays/src/Modal.tsx:42]: This seems weird that you are using the same ref for two components.

**Comment by devongovett** [comment]
I think this isn't needed anymore?

**Comment by snowystinger** [comment]
I don't remember most of the context at this point, but do we not want to upgrade react-transition-group? Still, happy to close, we can open an issue to do the upgrade at another point

**Comment by alkanna** [comment]
Hey, what's preventing us from getting this merged ?
This is a blocker for us currently :(

**Comment by dannify** [comment]
Related:  https://github.com/adobe/react-spectrum/pull/3291

**Comment by LFDanLu** [comment]
Handled in https://github.com/adobe/react-spectrum/pull/3865

**Pull request closed without merging**